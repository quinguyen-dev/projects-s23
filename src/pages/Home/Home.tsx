import { Link } from "react-router-dom";

export default function Home() {
    return (
        <>
            <nav>
                <h1>projects summer 2023</h1>
                <ul className="projects">
                    <li>
                        <Link to="/games">game state screen transitions</Link>
                    </li>
                    <li>
                        <Link to="/parallax">parallax</Link>
                    </li>
                    <li>
                        <Link to="/revolving-asset">revolving asset</Link>
                    </li>
                    <li>
                        <Link to="/html-render">phone render</Link>
                    </li>
                </ul>
            </nav>
        </>
    );
}
