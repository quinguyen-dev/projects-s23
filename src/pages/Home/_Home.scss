@use "styles/settings" as *;

.home {
  flex-direction: column;
}

.projects {
  @include reset-list;

  margin-top: 8px;

  li {
    padding: 6px 0;
  }

  a {
    @include center-content;

    justify-content: start;
    text-decoration: none;
  }

  a:before {
    background: $primary-font-color;
    content: "";
    flex: 1;

    height: 1.25px;
    max-width: 32px;

    transition: max-width 0.4s ease;
  }

  a:hover:before {
    max-width: 64px;
  }

  a:not(:hover) {
    opacity: 0.5;
  }

  span {
    padding-left: 12px;
  }
}
