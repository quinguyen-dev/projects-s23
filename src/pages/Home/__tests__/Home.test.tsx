import { render } from "@testing-library/react";
import Home from "../Home";
import { BrowserRouter } from "react-router-dom";

describe("Home Screen", () => {
  it("should render correctly", () => {
    const component = render(<Home />, { wrapper: BrowserRouter });
    expect(component).toMatchSnapshot();
  });
});
