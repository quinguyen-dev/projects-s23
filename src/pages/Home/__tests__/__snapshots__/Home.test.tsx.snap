// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Screen > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h1>
          projects summer 2023
        </h1>
        <ul
          class="projects"
        >
          <li>
            <a
              href="/games"
            >
              game state screen transitions
            </a>
          </li>
          <li>
            <a
              href="/parallax"
            >
              parallax
            </a>
          </li>
          <li>
            <a
              href="/revolving-asset"
            >
              revolving asset
            </a>
          </li>
          <li>
            <a
              href="/html-render"
            >
              phone render
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h1>
        projects summer 2023
      </h1>
      <ul
        class="projects"
      >
        <li>
          <a
            href="/games"
          >
            game state screen transitions
          </a>
        </li>
        <li>
          <a
            href="/parallax"
          >
            parallax
          </a>
        </li>
        <li>
          <a
            href="/revolving-asset"
          >
            revolving asset
          </a>
        </li>
        <li>
          <a
            href="/html-render"
          >
            phone render
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
